qiling/loader/elf.py

```diff
@@ -240,30 +238,17 @@ def load_with_ld(self, elffile: ELFFile, stack_addr: int, load_address: int, arg
 
             with open(interp_local_path, 'rb') as infile:
                 interp = ELFFile(infile)
+                min_vaddr = min(seg['p_vaddr'] for seg in interp.iter_segments(type='PT_LOAD'))
 
                 # determine interpreter base address
-                interp_address = int(self.profile.get('interp_address'), 0)
+                interp_address = int(self.profile.get('interp_address'), 0) if min_vaddr == 0 else 0
```

Comment:
Would appreciate a brief comment explaining the rationale the way `interp_address` is selected.